From bd803ea5750fa1ce9cb2c345976caf19b4866419 Mon Sep 17 00:00:00 2001
From: George Cherian <george.cherian@marvell.com>
Date: Fri, 26 Feb 2021 12:46:10 +0530
Subject: [PATCH 1266/1921] Revert "PCI: Add pci_iounmap"

This reverts commit 5e78843bdca51588ab92310aacdbcbf8e56c7389.

Change-Id: Ifb2e582479b7dc1b80478979635300e6b89a2161
Signed-off-by: George Cherian <george.cherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/46793
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 include/asm-generic/io.h | 4 ----
 lib/pci_iomap.c          | 9 ---------
 2 files changed, 13 deletions(-)

diff --git a/include/asm-generic/io.h b/include/asm-generic/io.h
index fd5eb8773846..b83e2802c969 100644
--- a/include/asm-generic/io.h
+++ b/include/asm-generic/io.h
@@ -891,16 +891,12 @@ static inline void iowrite64_rep(volatile void __iomem *addr,
 struct pci_dev;
 extern void __iomem *pci_iomap(struct pci_dev *dev, int bar, unsigned long max);
 
-#ifdef CONFIG_GENERIC_PCI_IOMAP
-extern void pci_iounmap(struct pci_dev *dev, void __iomem *p);
-#else
 #ifndef pci_iounmap
 #define pci_iounmap pci_iounmap
 static inline void pci_iounmap(struct pci_dev *dev, void __iomem *p)
 {
 }
 #endif
-#endif /* CONFIG_GENERIC_PCI_IOMAP */
 #endif /* CONFIG_GENERIC_IOMAP */
 
 /*
diff --git a/lib/pci_iomap.c b/lib/pci_iomap.c
index 36128af05e1c..2d3eb1cb73b8 100644
--- a/lib/pci_iomap.c
+++ b/lib/pci_iomap.c
@@ -134,13 +134,4 @@ void __iomem *pci_iomap_wc(struct pci_dev *dev, int bar, unsigned long maxlen)
 	return pci_iomap_wc_range(dev, bar, 0, maxlen);
 }
 EXPORT_SYMBOL_GPL(pci_iomap_wc);
-
-#ifndef CONFIG_GENERIC_IOMAP
-#define pci_iounmap pci_iounmap
-void pci_iounmap(struct pci_dev *dev, void __iomem *addr)
-{
-	iounmap(addr);
-}
-EXPORT_SYMBOL(pci_iounmap);
-#endif
 #endif /* CONFIG_PCI */
-- 
2.31.1

